### DISCUSSION TITLE
Reading the statistics

### DISCUSSION THREAD

**Discussion by dineshkumarreddyg36** [xcompact3d/Incompact3d#143] [Q&A] [answered]
Can someone tell me how to load the statistic files that are being written on giving "**initstat**" some value?

**Comment by mathrack** [comment, marked as answer]
Hello,

https://github.com/fschuch/xcompact3d_toolbox/
or
```
#! /usr/bin/env python3
#
import numpy as np
import argparse
#
parser = argparse.ArgumentParser(description='Print min / max and average in space.')
parser.add_argument('files', nargs='+', help='list of .bin files')
args = parser.parse_args()
#
# Incompact3d grid
#
nx = 128
ny = 65
nz = 64
f = open("yp.dat")
yp = np.fromfile(f, sep='\n')
f.close()
#
for binfile in args.files:
    # Read 3D binary file
    f = open(binfile, 'rb')
    data = np.fromfile(f) #, '<f4')
    f.close()
    # Print the file name and min / max
    print("File : " + binfile)
    print("   Min : " + str(np.min(data)))
    print("   Max : " + str(np.max(data)))
    # Map data in a 3D array
    array = np.reshape(data, (nx,ny,nz), order='F')
    # Compute spatial average and write a text file
    f = open(binfile+'.txt', 'w')
    for j in range(ny):
        val=np.sum(array[:,j,:])/np.float(nx*nz)
        f.write(str('{:.6e}'.format(yp[j])) + " " + str('{:.6e}'.format(val)) + '\n')
    f.close()

```

**Reply by dineshkumarreddyg36** [reply]
Hi,
Can someone suggest how to read the **turb-data** file in which **eddy viscosity** is being written and in what format is that data being written, whether is it time series data i.e., according to the **ioutput** value it adds the data to that file or something else. 
Thanks in advance.
Dinesh.

**Reply by pbartholomew08** [reply]
It should be writing out to separate files like ux, etc. However it appears to have a bug

**Reply by pbartholomew08** [reply]
https://github.com/xcompact3d/Incompact3d/issues/166

**Reply by pbartholomew08** [reply]
A fix is now available on pull request https://github.com/xcompact3d/Incompact3d/tree/166-turbulence-statistics-are-not-written-out-properly

Using this you should be able to treat the eddy viscosity files in the same way as ux, uy, etc. Although for visualisation you will need to write snapshot.xdmf files (you can adapt those written for the main flow variables for this).

**Reply by pbartholomew08** [reply]
This has now been merged, if you update to latest you should be able to output turbulent viscosity data per timestep

**Reply by dineshkumarreddyg36** [reply]
It's working now. 
Thank you
